tools/add_header

```diff
@@ -283,6 +283,75 @@ class PythonHeaderTemplate(HeaderTemplate):
         return ret
 
 
+class BashHeaderTemplate(HeaderTemplate):
+    @staticmethod
+    def _get_file_shebag(file_path: str) -> Optional[str]:
+        """
+        Returns the python file shebag.
+
+        :param file_path: The path to the file.
+        :returns: The python file shebag if present. This includes a new line character.
+        :throws UnicodeDecodeError: If the file could not be read.
+        """
+        with open(file_path, "r") as file_obj:
+            file_content = file_obj.readlines()
+
+        if len(file_content) == 0:
+            return None
+        if "#!/bin/bash" in file_content[0]:
+            return file_content[0]
+        return None
+
+    @staticmethod
+    def is_file_suitable(file_path: str) -> bool:
+        try:
+            is_file_using_shebag = BashHeaderTemplate._get_file_shebag(file_path) is not None
+        except UnicodeDecodeError:
+            # File could not be opened, this makes it automatically invalid
+            return False
+
+        return os.path.splitext(file_path)[1] == ".sh" or is_file_using_shebag
+
+    def __init__(self, file_path: str):
+        super().__init__(file_path)
+
+    def get_header(self) -> str:
+        file_creation_year = get_git_file_creation_year(self.file_path)
+
+        shebag_string = BashHeaderTemplate._get_file_shebag(self.file_path) or ""
+        copyright_string = "# Copyright CERN" \
```

Comment:
I would still suggest to use the full name like this:

European Organization for Nuclear Research (CERN)

instead of just the abbreviation

CERN


Also, file_creation_year will be wrong on some of them due to the way some of the files were moved around in git history rewrites. Wouldn't it be easier just to use the overall project duration and simply say "since 2012" everywhere?

Reply:
1. Yeah we can change that. I'd create a separate PR for this, since we also need to change this in the python files.
2. https://github.com/rucio/rucio/blob/47e93cfbe5887071c70de4ba815c1bbdddfac2ce/tools/add_header#L60 The `--follow` option in git should resolve this. This is why the year changes on some files. However we can set it fixed to 2012.